spec/providers/atmosphere/optimization_strategy/manual_spec.rb

```diff
@@ -11,13 +11,20 @@
   let(:cs) { create(:compute_site, active: true, funds: [fund]) }
   let(:at) { create(:filled_appliance_type, visible_to: 'all', preference_disk: 0) }
   let!(:tmpl) { create(:virtual_machine_template, appliance_type: at, compute_site: cs) }
-  let!(:fl1) { create(:virtual_machine_flavor, cpu: 1, memory: 512, compute_site: cs, active: true, hourly_cost: 1) }
-  let!(:fl2) { create(:virtual_machine_flavor, cpu: 2, memory: 1024, compute_site: cs, active: true, hourly_cost: 2) }
+  let!(:fl1) { create(:virtual_machine_flavor, cpu: 1, memory: 512, compute_site: cs, active: true) }
+  let!(:fl2) { create(:virtual_machine_flavor, cpu: 2, memory: 1024, compute_site: cs, active: true) }
   let(:cfg_tmpl) { create(:appliance_configuration_template, appliance_type: at)}
   let(:user) { create(:user) }
   let(:as) { create(:appliance_set, user: user)}
 
   it 'return templates and flavors for each defined vm in opt policy params' do
+
+    fl1.set_hourly_cost_for(at.os_family, 5)
+    fl2.set_hourly_cost_for(at.os_family, 10)
```

Comment:
The spec finds both flavors valid for at (even though they differ w.r.t. CPU and RAM), so it selects the cheaper flavor - which is nondeterministic because the flavor factory produces flavors with random cost unless cost is manually set (as done here). Omitting these lines causes random Fs.